eth-bytecode-db/eth-bytecode-db/src/verification/handlers/mod.rs

```diff
@@ -132,11 +177,49 @@ async fn process_verify_response(
         Ok(())
     };
 
+    let process_alliance_database_insertion = || async {
+        match alliance_db_action {
+            VerifierAllianceDbAction::IgnoreDb => {}
+            VerifierAllianceDbAction::SaveIfDeploymentExists {
+                db_client: alliance_db_client,
+                chain_id,
+                contract_address,
+                transaction_hash,
+            } => {
+                let database_source = DatabaseReadySource::try_from(source.clone())
+                    .context("Converting source into database ready version")?;
+
+                let deployment_data = db::verifier_alliance_db::ContractDeploymentData {
+                    chain_id,
+                    contract_address: contract_address.to_vec(),
+                    transaction_hash: transaction_hash.to_vec(),
+                };
+                db::verifier_alliance_db::insert_data(
+                    alliance_db_client,
+                    database_source,
+                    deployment_data,
+                )
+                .await
+                .context("Insert data into verifier alliance database")?;
+            }
+        }
+
+        Ok(())
+    };
+
     let _ = process_database_insertion()
         .await
         .map_err(|err: anyhow::Error| {
             tracing::error!("Error while inserting contract data into database: {err:#}")
         });
 
+    let _ = process_alliance_database_insertion()
```

Comment:
do we want to make `process_database` and `process_alliance_database` concurrenly?

however I see that this function`process_verify_response` is veryyy long and complex function now, so maybe it can be left without complications :)

Reply:
Updated :+1: